compound.hh

```diff
@@ -297,6 +301,28 @@ public:
         // FIXME: call equal() on each component
         return compare(v1, v2) == 0;
     }
+    bool has_memcmp_comparable_form() {
+        for (const auto& t : _types) {
+            if (t->is_reversed()) {
+                const auto& t2 = t->underlying_type();
+                if (!(t2 == bytes_type || t2 == utf8_type || t2 == long_type || t2 == ascii_type)) {
+                    return false;
+                }
+            } else if (!(t == bytes_type || t == utf8_type || t == long_type || t == ascii_type)) {
+                return false;
+            }
+        }
+        return true;
+    }
+    void memcmp_comparable_form(managed_bytes_view v, std::vector<std::byte>& out) {
```

Comment:
Why not return out?

Reply:
So it can append to an existing vector. (Which already contains e.g. the token).